engine/class_modules/sc_shaman.cpp

```diff
@@ -8536,6 +8635,26 @@ void shaman_t::create_buffs()
                             ->set_default_value_from_effect_type( A_ADD_PCT_MODIFIER, P_TICK_TIME )
                             ->apply_affecting_conduit( conduit.call_of_flame );
 
+  buff.fireheart = make_buff( this, "fireheart", spell.t28_2pc_ele )
+                       // TODO: confirm if duration is affected by conduit and tier set 4pc bonus
+                       ->set_duration( buff.fire_elemental->buff_duration() )
```

Comment:
Fireheart itself has a 30 second duration, this probably needs some checking if stuff is synced in game or not?

Reply:
Yes we're aware. That#s why the comment is right above this line. Once we have this on PR we need to do some testing.